code/Migration.php

```diff
@@ -14,4 +14,189 @@
+	 *
+	 * @param string $table
+	 * @param string $column
+	 * @return boolean
+	 */
+	protected static function tableColumnExists($table, $column) {
+		if (!self::tableExists($table)) return false;
+		$columns = self::getTableColumns($table);
+		return array_key_exists($column, $columns);
+	}
+
+	/**
+	 * Returns true if an array of columns exist on a database table
+	 *
+	 * @param string $table
+	 * @param array $columns
+	 * @return boolean
+	 */
+	protected static function tableColumnsExist($table, array $columns) {
+		if (!self::tableExists($table)) return false;
+		return count(array_intersect($columns, array_keys(self::getTableColumns($table)))) === count($columns);
+	}
+
+	/**
+	 * Returns an array of columns for a database table
+	 *
+	 * @param string $table
+	 * @return array (empty if table doesn't exist)
+	 */
+	protected static function getTableColumns($table) {
+		if (!self::tableExists($table)) return [];
+		return DB::fieldList($table);
+	}
+
+	/**
+	 * Drops columns from a database table.
+	 * Returns false if the table or any of the columns do not exist.
+	 * Returns true if the SQL query was executed.
+	 *
+	 * @param string $table
+	 * @param array $columns
+	 * @return boolean
+	 */
+	protected static function dropColumnsFromTable($table, array $columns) {
+		$queried = false;
+		$existingColumns = self::getTableColumns($table);
+		if ($existingColumns) {
+			foreach ($columns as $column) {
+				if (array_key_exists($column, $existingColumns)) {
+					DB::query("ALTER TABLE $table DROP COLUMN $column;");
+					$queried = true;
+				}
+			}
+		}
+		return $queried;
+	}
+
+	/**
+	 * Add columns to a database table if they don't exist.
+	 * Returns false if the table does not exist.
+	 * Returns true if the SQL query was executed.
+	 *
+	 * @param string $table
+	 * @param array $columns e.g. ['MyColumn' => 'VARCHAR(255) CHARACTER SET utf8']
+	 * @return boolean
+	 */
+	protected static function addColumnsToTable($table, array $columns) {
+		$queried = false;
+		$existingColumns = self::getTableColumns($table);
+		if ($existingColumns) {
+			foreach ($columns as $column => $properties) {
+				if (!array_key_exists($column, $existingColumns)) {
+					DB::query("ALTER TABLE $table ADD $column $properties;");
+					$queried = true;
+				}
+			}
+		}
+		return $queried;
+	}
+
+	/**
+	 * Gets the value for a single column in a row from the database by the ID column.
+	 * Useful when a field has been removed from the class' `$db` property,
+	 * and therefore is no longer accessible through the ORM.
+	 * Returns `null` if the table, column or row does not exist.
+	 *
+	 * @param string $table
+	 * @param string $columns
+	 * @param string||int $id
+	 * @return string
+	 */
+	protected static function getRowValueFromTable($table, $field, $id) {
+		$value = null;
+		if (self::tableColumnExists($table, $field)) {
+			$query = new SQLQuery();
+			$query->setFrom($table)->setSelect([$field])->setWhere("ID = $id");
```

Comment:
Would you mind casting this to an integer first? Unfortunately SS doesn't have parametrization yet. 2 other spots below too.